lib/routes/chainswap/receive/chainswap_qr_dialog.dart

```diff
@@ -64,9 +64,9 @@ class ChainSwapQrDialogState extends State<ChainSwapQrDialog> with SingleTickerP
   @override
   void didUpdateWidget(covariant ChainSwapQrDialog oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (widget.receiveOnchainResponse?.address != oldWidget.receiveOnchainResponse?.address) {
+    if (widget.receivePaymentResponse?.destination != oldWidget.receivePaymentResponse?.destination) {
       var inputCubit = context.read<InputCubit>();
-      inputCubit.trackPayment(widget.receiveOnchainResponse!.bip21).then((value) {
+      inputCubit.trackPayment(widget.receivePaymentResponse!.destination).then((value) {
```

Comment:
I think this needs checking. The destination is the BIP21 for a chain swap, but the `trackPayment` function is comparing it to the payment details swapId.

Reply:
Yes, Ross is right. We should compare the destination to the `Payment.destination` instead.